server/service.go

```diff
@@ -342,93 +336,70 @@ func (m *BoostService) handleGetHeader(w http.ResponseWriter, req *http.Request)
 	}
 }
 
-// respondPayload responds to the proposer with the payload
-func (m *BoostService) respondPayload(w http.ResponseWriter, log *logrus.Entry, result *builderApi.VersionedSubmitBlindedBlockResponse, originalBid bidResp) {
-	// If no payload has been received from relay, log loudly about withholding!
-	if result == nil || getPayloadResponseIsEmpty(result) {
-		originRelays := types.RelayEntriesToStrings(originalBid.relays)
-		log.WithField("relaysWithBid", strings.Join(originRelays, ", ")).Error("no payload received from relay!")
-		m.respondError(w, http.StatusBadGateway, errNoSuccessfulRelayResponse.Error())
-		return
-	}
-	m.respondOK(w, result)
-}
-
 // handleGetPayload requests the payload from the relays
 func (m *BoostService) handleGetPayload(w http.ResponseWriter, req *http.Request) {
-	log := m.log.WithField("method", "getPayload")
-	log.Debug("getPayload request starts")
+	var (
+		userAgent                        = wrapUserAgent(UserAgent(req.Header.Get(HeaderUserAgent)))
+		proposerContentType              = req.Header.Get(HeaderContentType)
+		proposerAcceptContentTypes       = req.Header.Get(HeaderAccept)
+		parsedProposerAcceptContentTypes = goacceptheaders.Parse(proposerAcceptContentTypes)
+		proposerEthConsensusVersion      = req.Header.Get(HeaderEthConsensusVersion)
```

Comment:
are there any assumptions here that these headers will be set? are they handled if empty?

Reply:
They are handled if they're empty. See the following:

`proposerContentType` will default to JSON if it's empty or invalid:

https://github.com/flashbots/mev-boost/blob/94f460400fb7abbdc9e4da79f9cdac0de4703ddd/server/get_payload.go#L50-L54

For accept media types, we will default to JSON too:

https://github.com/flashbots/mev-boost/blob/94f460400fb7abbdc9e4da79f9cdac0de4703ddd/server/service.go#L378-L389

For `proposerEthConsensusVersion`, if the content type is octet-stream and this is empty, we'll return an error.

https://github.com/flashbots/mev-boost/blob/94f460400fb7abbdc9e4da79f9cdac0de4703ddd/server/get_payload.go#L425-L427